lib/templatesimplifier.cpp

```diff
@@ -1318,6 +1318,9 @@ int TemplateSimplifier::getTemplateNamePosition(const Token *tok)
     else if (!getTemplateNamePositionTemplateFunction(tok, namepos))
         namepos = -1; // Name not found
     mTemplateNamePos[tok] = namepos;
+    if (mSettings->debugtemplate && it != mTemplateNamePos.end() && it->second != namepos) {
```

Comment:
This suggestion was only meant for your development testing.  I expected it to be removed once the new caching was proven to give the same results.

Reply:
I can remove it. I thought it might still be useful just in case I still missed a case.

Reply:
If the validation passes the clang test suite then there is probably no need for it.  The validation could be put in an assert if you are not confident it has been tested enough but the code doesn't make it easy to do that. 

I don't like overloading the debugtemplate setting for this purpose.